Network/Socket/SockAddr.hs

```diff
@@ -41,7 +45,27 @@ connect = G.connect
 -- 'defaultPort' is passed then the system assigns the next available
 -- use port.
 bind :: Socket -> SockAddr -> IO ()
-bind = G.bind
+bind s a = case a of
+  SockAddrUnix p -> do
+    -- gracefully handle the fact that UNIX systems don't clean up closed UNIX
+    -- domain sockets, inspired by https://stackoverflow.com/a/13719866
+    res <- try (G.bind s a)
+    case res of
+      Right () -> pure ()
+      Left e -> if not (isAlreadyInUseError (e :: IOException))
```

Comment:
Why don't you use `| isAlreadyInUseError (e :: IOException)` instead of `if`.

Reply:
Done in 8d5ff8d - though I've kept the error conditions grouped together at the top as I think that's visually easier to read, hope that's OK.

Reply:
Please don't forget `pure` in tests.

Reply:
woops, also done now in f27b87c